### PR TITLE
docs, model updates, add tests, add documentation

### PR DESCRIPTION
**Pull request by fivetran-reneeli**
## PR Overview
**This PR will address the following Issue/Feature:**
Adding conversion metrics 

**This PR will result in the following new package version:**

v0.4.0

breaking change to users who weren't bringing in conversion metrics already through the passthrough variable.

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

## Feature update: Conversions Support
- We have added conversion metrics to the end models. 
- The conversion metrics are the following:
  - `purchases_30_d`: Number of attributed conversion events occurring within 30 days of an ad click.
  - `sales_30_d`: Total value of sales occurring within 30 days of an ad click.

- In the event that you were already passing the above fields in via our [passthrough columns](https://github.com/fivetran/dbt_amazon_ads?tab=readme-ov-file#passing-through-additional-metrics), the package will dynamically avoid "duplicate column" errors.

> The above new field additions are 🚨 **breaking changes** 🚨 for users who were not already bringing in conversion fields via passthrough columns.

## Under the Hood
- Created `amazon_ads_persist_pass_through_columns` macro to ensure that the new conversion fields are backwards compatible with users who have already included them via passthrough fields.
- Added integrity and consistency validation tests within `integration_tests` folder for the transformation models (to be used by maintainers only).

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [ ] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [na] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [ ] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

Validation tests passing

### If you had to summarize this PR in an emoji, which would it be?
 
:dancer:

### CODE REVIEW

File: integration_tests/tests/integrity/vertical_sum_conversions
Comment (fivetran-joemarkiewicz) [inline integration_tests/tests/integrity/vertical_sum_conversions:1]: Is there a reason for this file? Should we remove it if it's empty.

File: packages.yml [hunk 1]
```diff
@@ -1,3 +1,7 @@
 packages:
-  - package: fivetran/amazon_ads_source
-    version: [">=0.3.0", "<0.4.0"]
+  # - package: fivetran/amazon_ads_source
+  #   version: [">=0.4.0", "<0.5.0"]
+
+  - git: https://github.com/fivetran/dbt_amazon_ads_source.git
+    revision: feature/conversions
+    warn-unpinned: false
```
Comment (fivetran-joemarkiewicz) [inline packages.yml]: Reminder to swap before release

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM

File: integration_tests/tests/integrity/vertical_sum_conversions
Comment (fivetran-jamie) [reply integration_tests/tests/integrity/vertical_sum_conversions:1]: Updated it with code to run vertical integrity tests on the new conversion metrics (with certain sources/model comparisons commented out due to our limited dataset). Test is passing 
image

**Pull request merged**